### DISCUSSION TITLE
Will there be an option for a custom ID?

### DISCUSSION THREAD

**Discussion by ozziestraw** [pocketbase/pocketbase#2518] [Q&A]
Currently have a separate column to store a custom ID, rather than the default `id` field -- but as we aren't allowed to update records via /api/collections/{collection}/records/:custom_id, will there be a possibility of setting the default ID to a custom value outside of 15 chars?

**Comment by ganigeorgiev** [comment]
The 15 characters limit of the `id` field will eventually be removed in the future but for now is a very low priority.

Could you elaborate a little more on your use case?

**Reply by ozziestraw** [reply]
Thanks for the reply!

So there is a route with a more user-friendly ID:
- example.com/{custom_id}

The issue with this is, I have to call the List endpoint first to retrieve the row with that custom ID, retrieve the actual ID from that and then make another call to Update

**Reply by tigawanna** [reply]
Pass it in an id field on create record and that will be the ID as long as it's long enough and unique

**Reply by ozziestraw** [reply]
It has to be 15 characters though unfortunately

**Reply by Lex-2008** [reply]
Yep, I agree that having an alternative way of specifying which record to update might be useful (like SQL's `UPDATE ... WHERE ...`), however it might complicate things if you happen to update several records at once, and PocketBase have to fire multiple update events to realtime API subscribers at once.

@Okozzie : I don't know details of your usage, but maybe your client application (or whatever uses PocketBase API) can save "pocketbase internal" record id at the same time it learns about custom id? After all, where does it get custom id from? Is it a user who types it from a keyboard? Or does your application call PocketBase API to get list of records / available IDs, which user later can open / view / edit / update? In that case, maybe you can store "pocketbase internal" record id next to your custom one. But again, it really depends on what you're doing, so I'm merely _suggesting_, and in no way telling you that it's the right way!

**Reply by bjornlll** [reply]
My use case is: I've got pocketbase running at the tail end of a simple ETL setup and a predictable identifier would be handy. I of course realize that I could add in an 'alt ID' that I use, instead, but would prefer to let my primary identifier be one that I can control.

**Reply by ganigeorgiev** [reply]
@bjornlll You are commenting on a very old discussion. With PocketBase v0.23+ users can change the `id` field from the related collection field options.

**Reply by bjornlll** [reply]
That's great to know! Does the 15 char restriction still apply? 

I was actually just reading through the release notes for PocketBase but didn't see it mentioned. Perhaps I missed it.

**Reply by jspablo** [reply]
Reading the docs I am not sure how to modify it, seems 15 char is still the limit https://pocketbase.io/docs/api-collections/#create-collection From the UI I see the option
